tests/e2e_test.go

```diff
@@ -0,0 +1,555 @@
+				Genesis:     warpGenesisFile,
+				ChainConfig: warpChainConfigPath,
+				SubnetSpec: &rpcpb.SubnetSpec{
+					SubnetConfig: "",
+					Participants: subnetANodeNames,
+				},
+			},
+			{
+				VmName:      evm.IDStr,
+				Genesis:     warpGenesisFile,
+				ChainConfig: warpChainConfigPath,
+				SubnetSpec: &rpcpb.SubnetSpec{
+					SubnetConfig: "",
+					Participants: subnetBNodeNames,
+				},
+			},
+		},
+	)
+	Expect(err).Should(BeNil())
+
+	// Issue transactions to activate the proposerVM fork on the receiving chain
+	fundedKey, err := crypto.HexToECDSA(fundedKeyStr)
+	Expect(err).Should(BeNil())
+	subnetB := manager.GetSubnets()[1]
+	subnetBDetails, ok := manager.GetSubnet(subnetB)
+	Expect(ok).Should(BeTrue())
+
+	chainBID := subnetBDetails.BlockchainID
+	uri := httpToWebsocketURI(subnetBDetails.ValidatorURIs[0], chainBID.String())
+
+	client, err := ethclient.Dial(uri)
+	Expect(err).Should(BeNil())
+	chainBIDInt, err := client.ChainID(ctx)
+	Expect(err).Should(BeNil())
+
+	err = utils.IssueTxsToActivateProposerVMFork(ctx, chainBIDInt, fundedKey, client)
+	Expect(err).Should(BeNil())
+
+	log.Info("Set up ginkgo before suite")
+})
+
+var _ = ginkgo.AfterSuite(func() {
+	log.Info("Running ginkgo after suite")
+	Expect(manager).ShouldNot(BeNil())
+	Expect(manager.TeardownNetwork()).Should(BeNil())
+	Expect(os.Remove(warpChainConfigPath)).Should(BeNil())
+	Expect(os.Remove(relayerConfigPath)).Should(BeNil())
+})
+
+// Ginkgo describe node that acts as a container for the relayer e2e tests. This test suite
+// will run in order, starting off by setting up the subnet URIs and creating a relayer config
+// file. It will then build the relayer binary and run it with the config file. The relayer
+// will then send a transaction to the source subnet to issue a Warp message simulting a transaction
+// sent from the Teleporter contract. The relayer will then wait for the transaction to be confirmed
+// on the destination subnet and verify that the Warp message was received and unpacked correctly.
+var _ = ginkgo.Describe("[Relayer E2E]", ginkgo.Ordered, func() {
+	var (
+		subnetIDs                        []ids.ID
+		subnetA, subnetB                 ids.ID
+		blockchainIDA, blockchainIDB     ids.ID
+		chainANodeURIs, chainBNodeURIs   []string
+		fundedKey                        *ecdsa.PrivateKey
+		err                              error
+		receivedWarpMessage              *avalancheWarp.Message
+		chainBWSClient                   ethclient.Client
+		chainARPCClient, chainBRPCClient ethclient.Client
+		chainARPCURI, chainBRPCURI       string
+		chainAIDInt                      *big.Int
+		chainBIDInt                      *big.Int
+		payload                          []byte
+		relayerCmd                       *exec.Cmd
+		relayerCancel                    context.CancelFunc
+	)
+
+	fundedKey, err = crypto.HexToECDSA(fundedKeyStr)
+	if err != nil {
+		panic(err)
+	}
+
+	ginkgo.It("Setup subnet URIs", ginkgo.Label("Relayer", "Setup"), func() {
```

Comment:
My understanding is that we'd want all common setup code (standing up the subnets, deploying Teleporter, etc.) to be in the `BeforeSuite` call. Individual tests can then live in their own `Describe` specs (AKA a suite), which can further be broken down into `It` specs. 

IIUC in an ordered suite `BeforeAll` is no different than the first `It` spec within the `Describe` suite.

Reply:
Cool, makes sense to me. That is my understand based on that documentation as well.

As is, just wanted to call out that `Setup subnet URIs` and `Deploy Teleporter Contract` are their own `It` specs rather than in the `BeforeSuite` implementation. I think that is fine for now though, as not all AWM relayer tests will necessarily depend on Teleporter moving forward.

Reply:
Updated to move those stages to `BeforeSuite` 👍